src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -76,6 +76,10 @@ private void OnApplicationStopping()
         private void OnProcessExit(object sender, EventArgs e)
         {
             ApplicationLifetime.StopApplication();
+            if(!_shutdownBlock.WaitOne(TimeSpan.FromMinutes(1)))
```

Comment:
@anurse how long would we really need to wait here? The default shutdown timeout is only 5s. The longer you wait the more likely someone will just kill it and never see the log.
Could we pull the HostOptions from DI and just use that timeout?
https://github.com/aspnet/Extensions/blob/ab1b0aa16453b7c5876365f5dc6d476153873af0/src/Hosting/Hosting/src/HostOptions.cs#L16

Reply:
Hrm... given that we have that timeout, I wonder if we should just block for that duration and then exit rather than this whole thing about logging and waiting indefinitely.

But yeah, at the least, we can use it as the timeout before logging.

@maryamariyan you can get access to the `HostOptions` object @Tratcher mentioned by adding a constructor parameter to `ConsoleLifetime` of type `IOptions<HostOptions>`. The DI system will automatically inject it for you and you can access properties on the `HostOptions` through that.